### ISSUE TITLE
Camera2Basic - Preview is stretched

### ISSUE THREAD

**Issue by guillaume-tgl** [android/camera-samples#256]
The Preview is stretched when running the Camera2Basic sample.

Device: Galaxy S9
Android version 10

![Screenshot_20200617-144518_Camera2Basic](https://user-images.githubusercontent.com/2244170/84899719-5594d400-b0a9-11ea-9259-573a4b95bafb.jpg)

![Screenshot_20200617-144512_Camera2Basic](https://user-images.githubusercontent.com/2244170/84899715-53327a00-b0a9-11ea-9e4e-a23d68871b11.jpg)

**Comment by ianyhlee** [comment]
Looks these samples are really buggy

**Comment by guillaume-tgl** [comment]
Is it the samples or the Camera2 API?...

**Comment by ianyhlee** [comment]
Not 100% sure, does below sample work correctly on your phone?
https://github.com/googlearchive/android-Camera2Basic
It should use camera2 API as well.

**Comment by guillaume-tgl** [comment]
This sample works, yes. I suspect it is because it uses a TextureView while this sample uses a SurfaceView but I'm not sure.

**Comment by ZTFtrue** [comment]
use this when ```surfaceCreated``` :
```
val la = Constraints.LayoutParams(1080, 1920);//  width , height , it neee you calculate
 viewFinder.setLayoutParams(la)
```
I guess becuse  ```AutoFitSurfaceView``` , ```onMeasure``` error.
I test  ```setAspectRatio  ```  , some times width and height  is reverse. if you  can know what is curretly, you can fix it.

**Comment by ZTFtrue** [comment]
I only  need  portrait,
i can use this:
```
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val width = MeasureSpec.getSize(widthMeasureSpec)
        val height = MeasureSpec.getSize(heightMeasureSpec)
        if (aspectRatio == 0f) {
            setMeasuredDimension(width, height)
        } else {

            // Performs center-crop transformation of the camera frames
            val newWidth: Int
            val newHeight: Int

            if (width < height ) {
                newHeight = (width*aspectRatio).roundToInt()
                newWidth = width;
            } else {
                newWidth = (height*aspectRatio).roundToInt()
                newHeight = height
            }

            Log.d(TAG, "Measured dimensions set: $newWidth x $newHeight")
            setMeasuredDimension(newWidth, newHeight)
        }
    }
```

**Comment by ilexbor** [comment]
Redmi Note 8 Pro preview is stretched

**Comment by snti** [comment]
The Preview is stretched too in my Pixel 3a XL.

**Comment by owahltinez** [comment]
It should now be fixed thanks to #281
